### ISSUE TITLE
[BUG] npm view {package} time returns versions which cannot be downloaded

### ISSUE THREAD

**Issue by fluffynuts** [npm/cli#2063]
### Current Behavior:
I'm writing a tool to "time-walk" to see what would have been downloaded (matching semvers) at a specific point in the past, specifically to try to track down which versions of some interdependent packages would unbreak a project I'm working on.

I find the cli `npm view {package} time` to be really useful, but have found that it can return versions which are not available, so I have to double-check availability of versions before attempting to install them. This may be intentional (perhaps the time info is simply a record of publications), or not, so I'm bringing this up here.

A good package to use to test this is `swiper`, which reports a version `5.4.2` that is not available at the registry.

### Expected Behavior:
I would expect versions reported in `npm view {package} time` to be available for download, though this may contradict expected behavior

### Steps To Reproduce:
1. `npm view swiper time`
2. note version 5.4.2 in the list
3. attempt to `npm install --no-save swiper@5.4.2` and get an error

### Environment:
OS: Windows 10 64-bit
Node: 12.18.0
NPM: 6.14.4

**Comment by ljharb** [comment]
I'm able to run `npm install --no-save swiper@5.4.2` just fine, and https://unpkg.com/browse/swiper@5.4.2/ seems to work as well. What error do you get when trying to install it?

**Comment by fluffynuts** [comment]
I _was_ getting something along the lines of "not able to find package swiper@5.4.2", but now, I'm not. It wasn't once-off either -- it was well repeatable! :confused:

**Comment by ljharb** [comment]
If you have an internal registry, or a proxy, it's possible that it was timing out or taking awhile to seed its own cache?

**Comment by fluffynuts** [comment]
We have a private registry (which supports producing), but I don't use it as the primary (rather use an .npmrc and namespacing) specifically to avoid oddness. No proxy either. 

I literally had to build verification into the tool I was working on (time-walker, up at npmjs, installs packages satisfying project semver as at specified date): my first pass just used `npm view {package} time` to pick a version, and later queried the registry to see if that version was listed in the versions collection because of the issue I was experiencing until I realised that time was in there too, so I refactored to only fetch json directly from the registry.

It's well odd. I know what I saw, multiple times, and can't repro now. I should have screenshot, because I probably sound a little mad.
